### PR TITLE
feat(route): 新增 bing 搜索; 迁移 bing 每日图片到 v2; 新增 百度搜索; 迁移 搜狗特色LOGO 到 v2 规范；添加 搜狗搜索；添加 Google Search

### PR DESCRIPTION
**Pull request by CaoMeiYouRen**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/bing
/bing/search/rss
/baidu/search/rss
/sogou/doodles
/sogou/search/rss
/google/search/rss
/google/search/rss/zh-CN,zh;q=0.9,en;q=0.8,ja;q=0.7
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
-   [x] New Route / 新的路由
  -   [x] Follows [v2 Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [v2 路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
-   [x] Documentation / 文档说明
  -   [ ] EN / 英文文档
  -   [x] CN / 中文文档
-   [ ] Full text / 全文获取
  -   [x] Use cache / 使用缓存
-   [x] Anti-bot or rate limit / 反爬/频率限制
  -   [x] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
-   [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  -   [x] Parsed / 可以解析
  -   [ ] Correct time zone / 时区正确
-   [ ] New package added / 添加了新的包
-   [ ] `Puppeteer`

### CODE REVIEW

File: website/docs/routes/other.mdx [hunk 1]
```diff
@@ -113,6 +113,10 @@ See [#app-store-mac-app-store](/routes/program-update#app-store-mac-app-store)
 
 </Route>
 
+## Bing 搜索
```
Comment (TonyRL) [inline website/docs/routes/other.mdx:116]: Please add a level 3 heading

**Review by TonyRL — COMMENTED** [review]
Invalid pubDate.

**Comment by CaoMeiYouRen** [comment]
`/bing/search` 和 `/so/search/rss` 在本地是可用的，我无法在本地复现这个问题。
我猜想可能是 ip 问题导致的，so.com 和 cn.bing.com 屏蔽了某些 ip 的访问。

**Comment by CaoMeiYouRen** [comment]
@TonyRL  Hello, could you please review this pull request?
If you would like me to commit it in separate batches, please provide feedback. Thank you.

File: lib/v2/google/radar.js [hunk 2]
```diff
@@ -1,6 +1,18 @@
 module.exports = {
     'google.com': {
         _name: '谷歌',
+        www: [
+            {
+                title: '搜索',
+                docs: 'https://docs.rsshub.app/routes/other#google-alerts',
```
Comment (TonyRL) [inline lib/v2/google/radar.js:7]:
```suggestion
                docs: 'https://docs.rsshub.app/routes/other#google',
```

File: lib/v2/google/search.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+const renderDescription = (description, images) => art(path.join(__dirname, './templates/description.art'), { description, images });
+
+module.exports = async (ctx) => {
+    const { keyword, language } = ctx.params;
+    const searchParams = new URLSearchParams({
+        q: keyword,
+    });
+    const _url = new URL('https://www.google.com/search');
+    _url.search = searchParams.toString();
+    const url = _url.toString();
```
Comment (TonyRL) [inline lib/v2/google/search.js:15]: `lowerCamelCase`
docs: https://docs.rsshub.app/joinus/advanced/script-standard

File: lib/v2/google/search.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        key,
+        async () =>
+            (
+                await got(url, {
+                    headers: {
+                        'Accept-Language': language,
+                    },
+                })
+            ).data
+    );
```
Comment (TonyRL) [inline lib/v2/google/search.js:27]: - Use the cache efficiently. Do not cache the whole HTML. Cache the processed output instead.

- Please also note that you're calling [`tryGet()`](https://docs.rsshub.app/joinus/advanced/use-cache#ctxcachetrygetkey-getvaluefunc--maxage--refresh-) without disabling `refesh`. The content will be kept extending its expiration time whenever it's called using the same key.

File: lib/v2/so/search.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+const { art } = require('@/utils/render');
+const path = require('path');
+
+const renderDescription = (description, images) => art(path.join(__dirname, './templates/description.art'), { description, images });
+
+module.exports = async (ctx) => {
+    const { keyword } = ctx.params;
+    const url = `https://www.so.com/s?q=${encodeURIComponent(keyword)}`;
+    const key = `360-so-search:${url}`;
+    const response = await ctx.cache.tryGet(key, async () => (await got(url)).data);
```
Comment (TonyRL) [inline lib/v2/so/search.js:12]: - Use the cache efficiently. Do not cache the whole HTML. Cache the processed output instead.

- Please also note that you're calling [`tryGet()`](https://docs.rsshub.app/joinus/advanced/use-cache#ctxcachetrygetkey-getvaluefunc--maxage--refresh-) without disabling `refesh`. The content will be kept extending its expiration time whenever it's called using the same key.

File: lib/v2/sogou/radar.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+module.exports = {
+    'sogou.com': {
+        _name: '搜狗搜索',
```
Comment (TonyRL) [inline lib/v2/sogou/radar.js:3]:
```suggestion
        _name: '搜狗',
```

File: lib/v2/sogou/search.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+const path = require('path');
+const { parseDate } = require('@/utils/parse-date');
+
+const renderDescription = (description, images) => art(path.join(__dirname, './templates/description.art'), { description, images });
+
+module.exports = async (ctx) => {
+    const { keyword } = ctx.params;
+    const url = `https://www.sogou.com/web?query=${encodeURIComponent(keyword)}`;
+    const key = `sogou-search:${url}`;
+    const response = await ctx.cache.tryGet(key, async () => (await got(url)).data);
```
Comment (TonyRL) [inline lib/v2/sogou/search.js:13]: - Use the cache efficiently. Do not cache the whole HTML. Cache the processed output instead.

- Please also note that you're calling [`tryGet()`](https://docs.rsshub.app/joinus/advanced/use-cache#ctxcachetrygetkey-getvaluefunc--maxage--refresh-) without disabling `refesh`. The content will be kept extending its expiration time whenever it's called using the same key.

Comment (CaoMeiYouRen) [reply lib/v2/sogou/search.js:13]: > * Use the cache efficiently. Do not cache the whole HTML. Cache the processed output instead.
> * Please also note that you're calling [`tryGet()`](https://docs.rsshub.app/joinus/advanced/use-cache#ctxcachetrygetkey-getvaluefunc--maxage--refresh-) without disabling `refesh`. The content will be kept extending its expiration time whenever it's called using the same key.

Ok, I will fix it

**Comment by CaoMeiYouRen** [comment]
> Successfully [generated](https://github.com/DIYgod/RSSHub/actions/runs/7264017898) as following:
> 
> http://localhost:1200/bing - Success ✔️
> http://localhost:1200/bing/search/rss - Success ✔️
> http://localhost:1200/baidu/search/rss - Success ✔️
> http://localhost:1200/so/search/rss - Failed ❌
> ```
> HTTPError: Response code 404 (Not Found)
> 
> Route requested: /so/search/rss
> Error message: MaxRedirectsError: Redirected 10 times. Aborting.
>     at Request._onResponseBase (/app/node_modules/.pnpm/got@11.8.6/node_modules/got/dist/source/core/index.js:859:35)
>     at Request._onResponse (/app/node_modules/.pnpm/got@11.8.6/node_modules/got/dist/source/core/index.js:948:24)
>     at ClientRequest.<anonymous> (/app/node_modules/.pnpm/got@11.8.6/node_modules/got/dist/source/core/index.js:962:23)
>     at Object.onceWrapper (node:events:629:26)
>     at ClientRequest.emit (node:events:526:35)
>     at origin.emit (/app/node_modules/.pnpm/@szmarczak+http-timer@4.0.6/node_modules/@szmarczak/http-timer/dist/source/index.js:43:20)
>     at HTTPParser.parserOnIncomingClient (node:_http_client:693:27)
>     at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)
>     at TLSSocket.socketOnData (node:_http_client:535:22)
>     at TLSSocket.emit (node:events:514:28)
> Helpful Information to provide when opening issue:
> Path: /so/search/rss
> Node version: v20.10.0
> Git Hash: 2944fab
> ```
> 
> ...

I don't know why 360 search always fails, but it is working fine in local testing.
```xml
<?xml version="1.0" encoding="UTF-8"?>
<rss xmlns:atom="http://www.w3.org/2005/Atom" version="2.0"


>
    <channel>
        <title>
            <![CDATA[rss - 360 搜索]]>
        </title>
        <link>https://www.so.com/s?q=rss</link>
        <atom:link href="http://localhost:1200/so/search/rss" rel="self" type="application/rss+xml" />
        <description>
            <![CDATA[rss - 360 搜索 - Made with love by RSSHub(https://github.com/DIYgod/RSSHub)]]>
        </description>
        <generator>RSSHub</generator>
        <webMaster>i@diygod.me (DIYgod)</webMaster>
        <language>zh-cn</language>
        <lastBuildDate>Tue, 19 Dec 2023 15:37:49 GMT</lastBuildDate>
        <ttl>5</ttl>
        <item>
            <title>
                <![CDATA[rss(简易信息聚合)百科]]>
            </title>
            <description>
                <![CDATA[简易信息聚合(也叫聚合内容)是一种RSS基于XML标准，在互联网上被广泛采用的内容包装和投递协议。RSS(Really Simple Syndication)是一种描述和同步网站内容的格式，是使用最广泛的XML应用。RSS搭建了信息迅速传播的一...

    
        <img src="https://so.360tres.com/dmfd/246_246_80/t01a2c734a908ae2546.webp" referrerpolicy="no-referrer">
    

]]>
            </description>
            <guid isPermaLink="false">https://baike.so.com/doc/5372688-5608634.html</guid>
            <link>https://baike.so.com/doc/5372688-5608634.html</link>
            <author>
                <![CDATA[baike.so.com]]>
            </author>
        </item>
        <item>
            <title>
                <![CDATA[RSS 教程| 菜鸟教程]]>
            </title>
            <description>
                <![CDATA[RSS 教程通过使用 RSS,您可以有选择地浏览您感兴趣的以及与您的工作相关的新闻。通过使用 RSS,您可以把需要的信息从不需要的信息(兜售信息,垃圾邮件等)中分离出来。...

    

]]>
            </description>
            <guid isPermaLink="false">https://www.so.com/link?m=ec6r8KB0fWWteO%2FbnbkdrIOd3EL9MqZgzZculkJkS83XTUIeoLtGb6EhV%2Ff6c0PyM0WdCqQDrYVBE%2FRPgY%2B692NBe%2FzwPirt0ex8jOt6zl4Zf%2F3BUGbGuTENumy3Abw4Z1lnF4A%3D%3D</guid>
            <link>https://www.so.com/link?m=ec6r8KB0fWWteO%2FbnbkdrIOd3EL9MqZgzZculkJkS83XTUIeoLtGb6EhV%2Ff6c0PyM0WdCqQDrYVBE%2FRPgY%2B692NBe%2FzwPirt0ex8jOt6zl4Zf%2F3BUGbGuTENumy3Abw4Z1lnF4A%3D%3D</link>
            <author>
                <![CDATA[www.runoob.com]]>
            </author>
        </item>
        <item>
            <title>
                <![CDATA[网易RSS订阅中心]]>
            </title>
            <description>
                <![CDATA[2020年10月9日&nbsp;-&nbsp;RSS是站点用来和其他站点之间共享内容的简易方式(也叫聚合内容)。RSS使用XML作为彼此共享内容的标准方式。它代表了Really Simple Syndication(或RDF S...

    

]]>
            </description>
            <guid isPermaLink="false">https://www.so.com/link?m=buYic69yWs6B6Pv%2F24zAoA7OaMVnmPFhq6o1ZraD5ZNKMWNpXoRbH0TgJEEanz4YRMam%2B5Nyzn%2BBeryR4h41gc0o%2BB2J7f6XGje4oEw%3D%3D</guid>
            <link>https://www.so.com/link?m=buYic69yWs6B6Pv%2F24zAoA7OaMVnmPFhq6o1ZraD5ZNKMWNpXoRbH0TgJEEanz4YRMam%2B5Nyzn%2BBeryR4h41gc0o%2BB2J7f6XGje4oEw%3D%3D</link>
            <author>
                <![CDATA[www.163.com]]>
            </author>
        </item>
        <item>
            <title>
                <![CDATA[rss - 知乎]]>
            </title>
            <description>
                <![CDATA[2021年3月18日&nbsp;-&nbsp;RSS(简易信息聚合,也叫Really Simple Syndication、聚合RSS、聚合内容),是一种消息来源格式规范,用以聚合经常发布更新数据的网站,例如博客...www.zhihu.com

    

]]>
            </description>
            <guid isPermaLink="false">https://www.so.com/link?m=z1SC8U4f6hSVWeD14FY796WKBZoPdiuhFOzo%2FxZL0ipxhj43i%2BWUUmuodSkUT3b1A2fCTF41lQAuVKu%2F7W4mCvsWpukPwC7v2BEqkH22AU5HsT6DS4Z6%2F4wTTq1w%3D</guid>
            <link>https://www.so.com/link?m=z1SC8U4f6hSVWeD14FY796WKBZoPdiuhFOzo%2FxZL0ipxhj43i%2BWUUmuodSkUT3b1A2fCTF41lQAuVKu%2F7W4mCvsWpukPwC7v2BEqkH22AU5HsT6DS4Z6%2F4wTTq1w%3D</link>
            <author>
                <![CDATA[www.zhihu.com]]>
            </author>
        </item>
        <item>
            <title>
                <![CDATA[你必读的RSS 订阅源有哪些? - 知乎]]>
            </title>
            <description>
                <![CDATA[2014年5月9日&nbsp;-&nbsp;如果您和我一样面临信息爆炸带来的困扰,请你尝试一下RSS方式的阅读生活。 我自己体验是:每天只需很少的时间,就能非常高效地阅读大量精华的信息,知识摄...

    

]]>
            </description>
            <guid isPermaLink="false">https://www.so.com/link?m=bYZyfhWo8nHO%2FqexzpLt4J%2FxW%2Ff8dHag39gIfb7a35VbFYduaZHjPnJJnrMSKl%2FJjEr1fjAHzx7e39XG9FptQ11QD6hPAOt4cAkXrBPUomAPnT%2Fx6d8XYwOpTDKbzt7An</guid>
            <link>https://www.so.com/link?m=bYZyfhWo8nHO%2FqexzpLt4J%2FxW%2Ff8dHag39gIfb7a35VbFYduaZHjPnJJnrMSKl%2FJjEr1fjAHzx7e39XG9FptQ11QD6hPAOt4cAkXrBPUomAPnT%2Fx6d8XYwOpTDKbzt7An</link>
            <author>
                <![CDATA[www.zhihu.com]]>
            </author>
        </item>
        <item>
            <title>
                <![CDATA[什么是RSS-中关村在线]]>
            </title>
            <description>
                <![CDATA[2013年1月6日&nbsp;-&nbsp;RSS(简易信息聚合,也叫聚合内容)是一种描述和同步网站内容的格式。RSS可以是以下三个解释的其中一个:ReallySimpleSyndication;RDF(Resou...news.zol.com.cn&gt;新闻中心

    

]]>
            </description>
            <guid isPermaLink="false">https://www.so.com/link?m=bEyRJJbW%2FMp1kbfXSAOojneosoOhtTnURAsSDEGgNzEXpTpjTvwxUD%2BK2OxaRLUM%2BDXQZnYMS%2Bqf7Hl14qgapWunIcllwbtxwYyuq%2BuSERu6FLXFO7NJzfJ4rOhGOXLoT</guid>
            <link>https://www.so.com/link?m=bEyRJJbW%2FMp1kbfXSAOojneosoOhtTnURAsSDEGgNzEXpTpjTvwxUD%2BK2OxaRLUM%2BDXQZnYMS%2Bqf7Hl14qgapWunIcllwbtxwYyuq%2BuSERu6FLXFO7NJzfJ4rOhGOXLoT</link>
            <author>
                <![CDATA[news.zol.com.cn&#62;新闻中心]]>
            </author>
        </item>
        <item>
            <title>
                <![CDATA[科学网—RSS]]>
            </title>
            <description>
                <![CDATA[RSS是站点用来和其他站点之间共享内容的一种简易方式(也叫聚合内容),通常被用于新闻和其他按顺序排列的网站,例如Blog。一段项目的介绍可能包含新闻的全部介绍等。或...

    

]]>
            </description>
            <guid isPermaLink="false">https://www.so.com/link?m=bzzgWRxx1ZCWbYzOllTSL8OgNqnZ3ptUH9lG5mEA%2B%2FcaXChRiXB%2Boz%2FFmG%2F65RxZ%2FyoqVI31ht%2BLcFUoJw446dJcXr8%2FD2cGNzwmpSpCmfVqkQFuzti5Kxw%3D%3D</guid>
            <link>https://www.so.com/link?m=bzzgWRxx1ZCWbYzOllTSL8OgNqnZ3ptUH9lG5mEA%2B%2FcaXChRiXB%2Boz%2FFmG%2F65RxZ%2FyoqVI31ht%2BLcFUoJw446dJcXr8%2FD2cGNzwmpSpCmfVqkQFuzti5Kxw%3D%3D</link>
            <author>
                <![CDATA[www.sciencenet.cn]]>
            </author>
        </item>
        <item>
            <title>
                <![CDATA[rss高清在线观看 - 360视频]]>
            </title>
            <description>
                <![CDATA[<img src="https://so.360tres.com/dmfd/340_226_80/t013fb2ac91c62e3c73.webp?size=500x281" referrerpolicy="no-referrer"><img src="https://so.360tres.com/dmfd/340_226_80/t015d9bae0bc3bb8abb.webp?size=500x281" referrerpolicy="no-referrer"><img src="https://so.360tres.com/dmfd/340_226_80/t01ba6701765ea36577.webp?size=1080x608" referrerpolicy="no-referrer"><img src="https://so.360tres.com/dmfd/340_226_80/t0154f3c85a2802677e.webp?size=1080x608" referrerpolicy="no-referrer"><img src="https://so.360tres.com/dmfd/340_226_80/t0185d83cdf11916542.webp?size=1080x608" referrerpolicy="no-referrer"><img src="https://so.360tres.com/dmfd/340_226_80/t014c92bcdd6f667319.webp?size=1080x608" referrerpolicy="no-referrer">
    

]]>
            </description>
            <guid isPermaLink="false">https://www.so.com/link?m=b44yP0N3%2FwqqZeHYeC25m%2BK8%2BWgvXL16DrK5uDHnZVpMtbQZSRxn6tYixIbs2%2FEkUgULaMkmUYYQt6gE1UdXiFpisHnFUt66roDndcUmXDEoXgzsVexcm8Nk6zrzW3Z7aAtRKo2hPRFerPKEQ1N2cxpBtCIAYxIo5kx8KHHpa%2FMTo1woJ7fY6Xzb%2FTck%3D</guid>
            <link>https://www.so.com/link?m=b44yP0N3%2FwqqZeHYeC25m%2BK8%2BWgvXL16DrK5uDHnZVpMtbQZSRxn6tYixIbs2%2FEkUgULaMkmUYYQt6gE1UdXiFpisHnFUt66roDndcUmXDEoXgzsVexcm8Nk6zrzW3Z7aAtRKo2hPRFerPKEQ1N2cxpBtCIAYxIo5kx8KHHpa%2FMTo1woJ7fY6Xzb%2FTck%3D</link>
        </item>
        <item>
            <title>
                <![CDATA[RSS - 高效率的阅读方式- 少数派]]>
            </title>
            <description>
                <![CDATA[RSS是什么RSS的全称是「简易内容聚合」(ReallySimpleSyndication),是一个能让你在一个地方订阅各种感兴趣网站的工具。一个网站支持RSS,就意味着每当它新发布一篇新...sspai.com

    

]]>
            </description>
            <guid isPermaLink="false">https://www.so.com/link?m=bFTqwWVhz9kct78l%2F61%2Bs%2Bmg4PiLXVd32HboYCE9q1rWxofv6Kdeo1UrDJbI3y1OiJKIXJMn%2F0k%2BpLu3TyP72BRhVKvjPN77%2FQ8OIWH4RHERGnoJD</guid>
            <link>https://www.so.com/link?m=bFTqwWVhz9kct78l%2F61%2Bs%2Bmg4PiLXVd32HboYCE9q1rWxofv6Kdeo1UrDJbI3y1OiJKIXJMn%2F0k%2BpLu3TyP72BRhVKvjPN77%2FQ8OIWH4RHERGnoJD</link>
            <author>
                <![CDATA[sspai.com]]>
            </author>
        </item>
        <item>
            <title>
                <![CDATA[RSS 是什么意思- 泪雪网]]>
            </title>
            <description>
                <![CDATA[2018年11月8日&nbsp;-&nbsp;RSS(Real Simple Syndication)是用于指代以标准化方式提供更新或共享信息的 Web Feed格式集合的术语。信息可以是网站或博客条目,新闻标题...www.leiue.com

    

]]>
            </description>
            <guid isPermaLink="false">https://www.so.com/link?m=eEDftZ1vRyxtcQUf%2FGQ5pZflefH%2FC0p3oL0zNEAAxVOP2jn77Vma0fL1%2BbOMaiPI6qjV%2Bg2IRsf7n%2FDgQIGIN3Ojx%2BFRBdqGeAnVHspf1B9iQynC99ulVzbwxJOY%3D</guid>
            <link>https://www.so.com/link?m=eEDftZ1vRyxtcQUf%2FGQ5pZflefH%2FC0p3oL0zNEAAxVOP2jn77Vma0fL1%2BbOMaiPI6qjV%2Bg2IRsf7n%2FDgQIGIN3Ojx%2BFRBdqGeAnVHspf1B9iQynC99ulVzbwxJOY%3D</link>
            <author>
                <![CDATA[www.leiue.com]]>
            </author>
        </item>
    </channel>
</rss>
```

File: lib/v2/baidu/search.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+                            title: element.find('h3').first().text(),
+                            description: renderDescription(description, imgs),
+                            link: element.find('h3 a').first().attr('href'),
+                            author: element.find('.c-row .c-color-gray').first().text() || '',
+                        };
+                    }
+                    return null;
+                })
+                .toArray()
+                .filter((e) => e && e.link);
```
Comment (TonyRL) [inline lib/v2/baidu/search.js:42]:
```suggestion
                .filter((e) => e?.link);
```

File: lib/v2/baidu/search.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        },
+        config.cache.contentExpire,
```
Comment (TonyRL) [inline lib/v2/baidu/search.js:44]:
```suggestion
        config.cache.routeExpire,
```

File: lib/v2/google/search.js [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+                    const author = element.find('div > div > div > div > div > span > a > div > div > span').first().text() || '';
+                    return {
+                        link,
+                        title,
+                        description: renderDescription(description, imgs),
+                        author,
+                    };
+                })
+                .toArray()
+                .filter((e) => e && e.link);
```
Comment (TonyRL) [inline lib/v2/google/search.js:50]:
```suggestion
                .filter((e) => e?.link);
```

File: lib/v2/google/search.js [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+        },
+        config.cache.contentExpire,
```
Comment (TonyRL) [inline lib/v2/google/search.js:52]:
```suggestion
        config.cache.routeExpire,
```

File: lib/v2/so/search.js [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+                    return {
+                        link: element.find('h3 a').first().attr('href'),
+                        title: element.find('h3').first().text(),
+                        description: renderDescription(description, imgs),
+                        author: element.find('.g-linkinfo cite').first().text() || '',
+                    };
+                })
+                .toArray();
+        },
+        config.cache.contentExpire,
```
Comment (TonyRL) [inline lib/v2/so/search.js:37]:
```suggestion
        config.cache.routeExpire,
```

File: lib/v2/sogou/search.js [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+                        title,
+                        description: renderDescription(description, imgs),
+                        author,
+                        pubDate,
+                    };
+                })
+                .toArray()
+                .filter((e) => e.link);
+        },
+        config.cache.contentExpire,
```
Comment (TonyRL) [inline lib/v2/sogou/search.js:44]:
```suggestion
        config.cache.routeExpire,
```

File: lib/v2/so/search.js [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+const renderDescription = (description, images) => art(path.join(__dirname, './templates/description.art'), { description, images });
+
+module.exports = async (ctx) => {
+    const { keyword } = ctx.params;
+    const url = `https://www.so.com/s?q=${encodeURIComponent(keyword)}`;
+    const key = `360-so-search:${url}`;
+    const items = await ctx.cache.tryGet(
+        key,
+        async () => {
+            const response = (await got(url)).data;
```
Comment (TonyRL) [inline lib/v2/so/search.js:16]: This request requires cookie.
https://github.com/DIYgod/RSSHub/actions/runs/7309647602/job/19917459502#step:12:35

Comment (CaoMeiYouRen) [reply lib/v2/so/search.js:16]: > This request requires cookie. https://github.com/DIYgod/RSSHub/actions/runs/7309647602/job/19917459502#step:12:35

Thank you, I will fix this issue as soon as possible

Comment (CaoMeiYouRen) [reply lib/v2/so/search.js]: > cookie.

Hello, 
I have tried several ways to solve the issue, but I still cannot pass the GitHub Action request. 
I tested it locally and did not encounter the MaxRedirectsError or cookie problem. Even when accessing it through a proxy, I did not encounter this issue. 
Therefore, I suspect it is an IP issue with GitHub Action, and there may be some restrictions on so.com. 
If it doesn't work, I will temporarily remove the route to so.com to ensure that other changes can be merged in.

**Pull request merged**